### PR TITLE
Add useful headers

### PR DESCRIPTION
**Pull request by dorsha**
## Description
Add useful headers to each request

## Must
- [x] Tests
- [ ] Documentation (if applicable)

### CODE REVIEW

File: descope/api/client.go [hunk 1]
```diff
@@ -440,3 +447,17 @@ func (c *Client) parseResponseError(response *http.Response) error {
...
 	return response.StatusCode >= http.StatusOK && response.StatusCode < http.StatusMultipleChoices || response.StatusCode == http.StatusTemporaryRedirect
 }
+
+func (c *Client) addDescopeHeaders(req *http.Request) {
+	req.Header.Set("x-descope-sdk-name", "golang")
+	req.Header.Set("x-descope-sdk-go-version", runtime.Version())
+	if c.buildInfo != nil {
+		for _, dep := range c.buildInfo.Deps {
+			if strings.HasPrefix(dep.Path, "github.com/descope/go-sdk/descope") {
+				req.Header.Set("x-descope-sdk-version", dep.Version)
```
Comment (orius123) [inline descope/api/client.go:457]: Need to ask devops to add these headers

**Review by orius123 — APPROVED** [review]

File: descope/api/client.go (hunk 1 above)
Comment (dorsha) [reply descope/api/client.go:457]: yes

File: descope/api/client.go [hunk 2]
```diff
@@ -370,6 +376,7 @@ func (c *Client) DoRequest(method, uriPath string, body io.Reader, options *HTTP
 		bearer = fmt.Sprintf("%s:%s", bearer, pswd)
 	}
 	req.Header.Set(AuthorizationHeaderName, BearerAuthorizationPrefix+bearer)
+	c.addDescopeHeaders(req)
```
Comment (asafshen) [inline descope/api/client.go:379]: theoretically we can calculate this once and put it on a private member, right?

**Review by asafshen — APPROVED** [review]

File: descope/api/client.go (hunk 2 above)
Comment (dorsha) [reply descope/api/client.go:379]: yes, the heavy part is done only once

**Pull request merged**